src/test/java/org/candlepin/model/PoolTest.java

```diff
@@ -295,4 +303,128 @@ public void testIsNotDerivedPool() {
 
         assertFalse(pool.isDerived());
     }
+
+    @Nested
+    @DisplayName("Field Access Tests")
+    public class FieldAccessTest extends DatabaseTestFixture {
+
+        @Test
+        public void testGetProductUuidWithDesyncedProduct() throws Exception {
+            Owner owner = this.createOwner();
+            Product product = this.createProduct(TestUtil.randomString("id-"),
+                TestUtil.randomString("name-"),
+                owner);
+            Pool pool = this.createPool(owner, product);
+
+            // We don't want to persist the upcoming changes to the pool, so commit now
+            this.getCurrentSession().getTransaction().commit();
```

Comment:
You can use `this.commitTransaction()` here, utilizing the declaration from `DatabaseTestFixture` instead of having to fetch the session manually.

Also, in general, we should use JPA interfaces over Hibernate interfaces wherever reasonably possible.

Reply:
Updated! Also, it is good to know to use the JPA interfaces over the Hibernate ones. I was honestly not sure which one to use.